### ISSUE TITLE
The B M & T of equalizer is being cut off -- Qt6

### ISSUE THREAD

**Issue by sharmanr** [mpogue2/SquareDesk#736]
When I build SquareDesk the bottom of the B, M and T of the 3-band EQ is now cut off.  The group is either too tall or is positioned too high.

Previously this was not the case, so I assumed this was because of a recent change.  However, I don't think it has changed, so perhaps it is acting differently under Qt6 compared with Qt5?.

For me, the attached patch makes all the the letters are visible.

[patch.eqsize.txt](https://github.com/mpogue2/SquareDesk/files/10149547/patch.eqsize.txt)

**Comment by mpogue2** [comment]
Without this change:

![image](https://user-images.githubusercontent.com/6316003/205812878-98690b56-0c22-4c58-8c85-ad04f6bd6fdd.png)

With this change:

![image](https://user-images.githubusercontent.com/6316003/205813088-60c6d9c5-1cf9-40e9-83d6-92ce513a3144.png)

**Comment by mpogue2** [comment]
After changing the EQBottomSpacer to 1px high, this fixes it for me, whereas the originally proposed patch makes it worse for me.

![image](https://user-images.githubusercontent.com/6316003/205815464-e48a33e4-af24-4de2-aa5e-30b5626f4412.png)

**Comment by mpogue2** [comment]
I'm now on Ventura and Qt6.4.1, so maybe one of those makes a difference (most likely Qt6.4.1)?
Could you try upgrading to 6.4.1, with this latest version (top of the master branch, 5721974f9fb85609e754922e80f0bfb44b0bc07f), and post here what you see?

**Comment by mpogue2** [comment]
BTW, if you take a screenshot with Cmd-Shift-5, and set Options to "Clipboard", and hit ENTER to take the screenshot, you will be able to paste that screenshot directly into the GitHub Write box just by hitting Cmd-V.

**Comment by sharmanr** [comment]
I added a comment but it seems to have disappeared, so I'll try again; sorry if this is a duplicate. 

Screen shot before either patch (macOS Monterey 12.6, Qt 6.4);  note the bottom part of B, M & T is cut off:
<img width="196" alt="ss-original" src="https://user-images.githubusercontent.com/76225465/205823485-f8bec20c-fc7f-4f10-8da8-82de9f9950ae.png">
Screenshot after my patch applied (B, M & T appear ok)

<img width="196" alt="ss-with-my-fix" src="https://user-images.githubusercontent.com/76225465/205823591-330c740c-1da4-441d-b661-e6dd8f57921e.png">
Screenshot with Mike's fix [setting height to 1], the letters disappear:

<img width="196" alt="ss-with-mikes-fix" src="https://user-images.githubusercontent.com/76225465/205823662-86d4ab80-70f8-4075-b004-83a2c4697ddc.png">

**Comment by mpogue2** [comment]
Hmm...well that's unexpected!  Are you compiling the top-of-tree, or something further down from the top?

**Comment by sharmanr** [comment]
I thought I was up to date.  I can take another copy to double check.

**Comment by mpogue2** [comment]
OK, I figured out the problem.  Give me a minute to describe it.  Both of our fixes are incorrect. :-)

**Comment by mpogue2** [comment]
The problem is actually NOT the height of the EQ box, nor is it the height of the EQ spacer.  It's the Zoom level, which is different between us.  I think we are seeing different results, because I always zoom way in, and you don't.

Default zoom level, top-of-tree, Qt6.4.1, Ventura:
![image](https://user-images.githubusercontent.com/6316003/205825830-62d793c5-1951-41ca-aaee-195a64b3a014.png)

Zoomed in 5X:
![image](https://user-images.githubusercontent.com/6316003/205826080-69354104-f933-4e41-b9bb-f0db7215987a.png)

Zoomed in all the way (press Zoom In 6X):
![image](https://user-images.githubusercontent.com/6316003/205825995-67810d0e-a06d-496d-b3af-1816e7628a5f.png)

**Comment by sharmanr** [comment]
Ah,  I haven't used Zoom for a while.  Well. not that Zoom anyway...

**Comment by mpogue2** [comment]
I'm not sure how to fix this.  But, I notice that the height of the sliders is the same in all of the Zoom levels above.  
So, I suspect that when the Tempo/Pitch/Volume/Mix text gets big at high Zoom levels, it makes the EQ section taller, which allows the B/M/T to show up.

Maybe we need to make the B/M/T sliders auto sized?

**Comment by sharmanr** [comment]
I'm out of my depth now!   (It's my first time in Qt so I can't really comment on it!!!)

I downloaded a new copy of the source just now, did the soundtouch thing, but building gets an error:
Could not resolve SDK SDKVersion for 'macosx13.0' using --show-sdk-version
then build stops sub-test123-make_first Error 3

**Comment by sharmanr** [comment]
Sorry,  changing source wasn't good;  I need to reload project so it configures it. I think.  
Anyway, it builds now and I can confirm that Zooming changes it as you say.
Making the sliders auto size sounds like a good idea, though I don't know how to do that [yet]!

**Comment by mpogue2** [comment]
I tried my best to make them autosizing, but no joy.  There probably is a way to do it with some sizing settings, but I'm gonna try to fallback to the same mechanism used for fonts, that is, manual size setting based on zoom setting.

**Comment by sharmanr** [comment]
I thought one of the advantages of containers in Qt was that they auto-sized automatically.  Maybe that’s just the spacing between things that’s automatic.

**Comment by mpogue2** [comment]
You're right they are SUPPOSED to auto-size.  Problem is that the QGridLayout (which holds the sliders and their labels) inside the QGroupBox that groups controls into the "3-band EQ" section doesn't seem to resize itself, no matter what I try.  This could be a bug in Qt, or maybe I'm doing it wrong.  But, I *think* I'm doing it right. :-)

I can't even control the size of these things *manually* (via the Zoom In/Out part of the code that carefully controls all the font sizes as the zoom changes), and I've tried everything I can think of so far...

**Comment by mpogue2** [comment]
It turns out that it's the QGroupBox that is preventing proper resizing behavior.  For now, I've removed the QGroupBox from the hierarchy.  

Commit: 0dfb6c1db2dae59d4a3e6170ae829da581e45c13

Please check this out on your laptop, and verify that it looks OK!  We lose the "3-band EQ" text, but we gain back auto resizing.  I'd rather have it resize properly at all zoom levels, if we are forced to choose.

**Comment by mpogue2** [comment]
With this change on my laptop, it looks like this at Zoom 0 (default zoom):
![image](https://user-images.githubusercontent.com/6316003/205860628-90efd457-4fef-4ed9-9a24-34df6351c5c2.png)

and like this at max zoom:
![image](https://user-images.githubusercontent.com/6316003/205860738-a92cd9cb-16ca-4da7-a7e8-fd93a9777b14.png)

**Comment by sharmanr** [comment]
Sorry, but this isn't working for me.  If I zoom in all is well:
<img width="192" alt="zoomed-in" src="https://user-images.githubusercontent.com/76225465/206061320-492e565a-946b-4bba-8237-56504a7bf38c.png">
but if I zoom out we lose the letters
<img width="199" alt="zoomed-out" src="https://user-images.githubusercontent.com/76225465/206061395-c5bf3092-95b1-4b70-857f-3d976382efd0.png">

I regret raising this issue!   I thought maybe I'd messed up the change, so downloaded the whole source but that's the behaviour I guess.

**Comment by mpogue2** [comment]
Note that the comment is actually incorrect (I should fix that at some point).  It currently is requiring 12.1 for the X86_64 build, and 13.0 for the ARM64 build.

**Comment by sharmanr** [comment]
Thank you.  Success!  I found the setting in test123.pro (I had changed it in the other .pro file to no avail!).  Running xcodebuild -showsdks showed macosx13.0 (which surprised me since I'm running 12.6) -- so I didn't have to change the one in test123 in the end.
The reason the new source code seemed to be the same as before was that I had renamed the directory before downloading it, but the Finder window remained open and was "following" this renamed directory, so when I double-clicked on what I thought was the new SquareDesk.pro it was in fact the old one.  A stupid mistake that took a long time to find!
